test/prompt/test_prompt_node.py

```diff
@@ -1228,3 +1228,29 @@ def test_chatgpt_direct_prompting_w_messages(chatgpt_prompt_model):
 
     result = pn(messages)
     assert len(result) == 1 and all(w in result[0].casefold() for w in ["arlington", "texas"])
+
+
+@pytest.mark.integration
+def test_content_moderation_gpt_3_5(chatgpt_prompt_model):
```

Comment:
Let's change these two tests so that they become unit tests. One test could check that the request to OpenAI is prepared correctly and the other test could check that a mocked response is processed correctly and results in an empty list of results. 
The reason is that we don't want to test the OpenAI api (also because it's unavailable from time to time). So you can assume it returns the expected response and mock any api calls via the openai_request method. You can find other tests mocking the openai request by searching for `@patch("haystack.utils.openai_utils.requests")` in the code base.

Reply:
That is a good point. It is quite hard to get ChatGPT to violate the moderation endpoint anyways so that case will also be testable when mocked.

Reply:
I reimplemented the tests to mock all uses of the OpenAI API. I added a unit test for `check_openai_policy_violation` in `test_openai_utlis` where I mocked the moderation endpoint response and a combined test for the call of PromptNode for both `text-davinci-003` and `gpt-3.5` where I mocked both endpoints (moderation & completion) and simulated the possible cases of moderation checks passing / failing.